app/jobs/reports/monthly_key_metrics_report.rb

```diff
@@ -24,6 +25,7 @@ def perform(date = Time.zone.yesterday.end_of_day)
       end
 
       reports.each do |report|
+        next if Rails.env.development? # we don't have s3 access locally
         upload_to_s3(report.table, report_name: report.filename)
```

Comment:
if we want this kind of bail logic (which I am not sure we do) it should go inside the shared `upload_to_s3` method on the base report

Reply:
Agreed that this isn't the right place.

I included this to run the job again locally tonight. With the parallel code added back, the new class completed in something like 15-20 minutes. (Which is about what you'd expect, having the 3 queries run in parallel.)

I don't feel strongly about whether this should be included at all. If we do include it, it should probably also log something vs. just silently skipping over it. The argument for including it is that it was _very_ annoying to have the job fail the first time hitting a 1-hour timeout, and then to succeed in just under an hour the second time, but then choke on the S3 upload. But this is sufficiently uncommon that it's possible no one will ever run into this again.